lighthouse-core/test/audits/legacy-javascript-test.js

```diff
@@ -83,6 +83,18 @@ describe('LegacyJavaScript audit', () => {
     assert.equal(result.extendedInfo.signalCount, 0);
   });
 
+  it('passes code with a legacy polyfill in third party resource', async () => {
+    const artifacts = createArtifacts([
+      {
+        code: 'String.prototype.repeat = function() {}',
+        url: 'https://www.googletagmanager.com/a.js',
```

Comment:
would be nice to ensure the main entity logic is getting exercised too

Reply:
I see us re-testing this particular logic over and over again ... maybe just one test in the `lib` that this fn will move to?

and just assume that usage within audits is OK (treat like a dependency, for example we don't test that the `URL` class works as expected in every test)

Reply:
Make the `mainEntity` a required argument, and you've got a deal :) 

making it required+typechecking solves exactly my concern, and agreed it would like testing URL at that point.